packages/clarity-js/src/data/metadata.ts

```diff
@@ -108,17 +122,16 @@ function shortid(): string {
 }
 
 function session(): Session {
-  let output: Session = { id: shortid(), ts: Math.round(Date.now()), count: 1, upgrade: BooleanFlag.False, upload: Constant.Empty };
-  if (config.track && supported(window, Constant.SessionStorage)) {
-    let value = sessionStorage.getItem(Constant.StorageKey);
-    if (value && value.indexOf(Constant.Separator) >= 0) {
-      let parts = value.split(Constant.Separator);
-      if (parts.length === 5 && output.ts - num(parts[1]) < Setting.SessionTimeout) {
-        output.id = parts[0];
-        output.count = num(parts[2]) + 1;
-        output.upgrade = num(parts[3]);
-        output.upload = parts[4];
-      }
+  let output: Session = { session: shortid(), ts: Math.round(Date.now()), count: 1, upgrade: BooleanFlag.False, upload: Constant.Empty };
+  let value = getCookie(Constant.SessionKey);
+  if (value) {
+    let parts = value.split(Constant.Separator);
+    if (parts.length === 5 && output.ts - num(parts[1]) < Setting.SessionTimeout) {
+      output.session = parts[0];
```

Comment:
>session [](start = 13, length = 7)

how are we dealing with overlapping pageviews across tabs?

Reply:
It's handled in the visualization right now.